### DISCUSSION TITLE
How can I make authorized API calls when loading content?

### DISCUSSION THREAD

**Discussion by ungario** [GrapesJS/grapesjs#4857] [Q&A] [answered]
My storage manager is configured in the following way:
```
storageManager: {
			autosave: true,
			type: 'remote',
			stepsBeforeSave: 1,
			options: {
				remote: {
					urlStore: `${templateEndpoint}${id!}`,
					urlLoad: `${templateEndpoint}${id!}`,
					onStore: templateSaveProcessor,
					onLoad: templateLoadProcessor,
					fetchOptions: (opts: object) => {
						const optsConv = opts as { method: string }
						return {
							method: optsConv.method === 'POST' ? 'PUT' : 'GET',
						}
					},
				},
			},
		},
```

The endpoints require authentication and a way of handling the situation when a new auth token has to be generated - for instance if the fetch cannot be performed due to an expired auth token. So is there any way to change the way how grapesjs does the fetch?

**Comment by artf** [comment, marked as answer]
You can [replace the remote storage](https://grapesjs.com/docs/modules/Storage.html#replace-storage) with your own implementation and do whatever logic you need.
